common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -447,28 +442,50 @@ private UTF8String toUpperCaseSlow() {
     return fromString(toString().toUpperCase());
   }
 
+  /**
+   * Optimized lowercase comparison for UTF8_BINARY_LCASE collation
```

Comment:
Can you describe in comment what lowercase comparison means?
Maybe better name is compareCaseInsensitive?

Reply:
That's not exactly equivalent to case-insensitive comparison. For example, `"ﬀ".compareLowerCase("ff") == false`, but case-insensitive should return true because uppercase converts both strings to "FF". Added clarification to the comment.